### PR TITLE
Add Ember integration

### LINKED ISSUE #14: Ember integration

### PR DESCRIPTION
**Pull request by xadamy**
Closes #14.

### CODE REVIEW

File: packages/ember/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+export function setupAppsignal(appsignal: any, Ember = (window as any).Ember) {
+  const prevHandler = Ember.onerror
+
+  if (!appsignal) {
+    throw new Error("No instance of Appsignal supplied.")
```
Comment (tombruijn) [inline packages/ember/src/index.ts]: what's the advantage here of throwing an error over logging an error and returning early from the function?

File: packages/ember/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    const span = appsignal.createSpan((span: any) =>
+      span.setAction("Ember.onerror").setError(error)
+    )
+
+    appsignal.send(span)
+
+    if (prevHandler && typeof prevHandler === "function") {
+      prevHandler.apply(this, arguments as any)
+    } else if (Ember.testing) {
+      throw error
```
Comment (tombruijn) [inline packages/ember/src/index.ts]: Does this mean we only rethrow the error in testing? Why not in production?

File: packages/ember/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  Ember.RSVP.on("error", function(reason: any): void {
+    const span = appsignal.createSpan((span: any) =>
+      span.setAction("Ember.RSVP.on")
+    )
+
+    if (reason instanceof Error) {
+      span.setError(reason)
+    } else {
+      span.setError({
+        name: "UnhandledPromiseRejectionError",
```
Comment (tombruijn) [inline packages/ember/src/index.ts]: Not sure what this error name comes from? Is it always a promise rejection?

**Comment by tombruijn** [comment]
Didn't mean to reject here. Just have some questions.

File: packages/ember/src/index.ts (hunk 3 above)
Comment (xadamy) [reply packages/ember/src/index.ts]: It's always an unhandled promise rejection, this is a listener that is fired when Ember's internal promise implementation throws an unhandled exception.

File: packages/ember/src/index.ts (hunk 2 above)
Comment (xadamy) [reply packages/ember/src/index.ts]: This is so any assertions dependent on throwing are triggered in testing mode. We don't normally rethrow in the other integrations, so this is for consistency.

File: packages/ember/src/index.ts (hunk 1 above)
Comment (xadamy) [reply packages/ember/src/index.ts]: Fixed in 8ec5cef.

**Comment by xadamy** [comment]
@tombruijn @thijsc I have removed the throws in favour of logging and returning early 👍

File: packages/ember/src/index.ts (hunk 3 above)
Comment (tombruijn) [reply packages/ember/src/index.ts]: ah maybe that's a good thing to add as a code comment?

Comment (xadamy) [reply packages/ember/src/index.ts]: Fixed in d0a0024.

**Comment by timonv** [comment]
This  is amazing! <3

**Pull request merged**